cpp/CMakeLists.txt

```diff
@@ -889,36 +896,50 @@ add_subdirectory(src/arrow/ipc)
 
```

Comment:
It is okay to install the headers for now. Parquet `FindArrow.cmake` looks for `arrow/arrow.h`